refactor(block-editor): Make indent/outdent functionality works with paragraph / lists

oidacra opened the pull request:
This pull request introduces an `IndentExtension` to the block editor, enhancing text formatting capabilities by adding support for indenting and outdenting paragraphs, headings, and blockquotes. It also includes minor code cleanups and reorganization of imports. Below are the key changes:

### New Feature: Indentation Support
* Added a new `IndentExtension` to handle indentation for paragraphs, headings, and blockquotes, including commands (`indent`, `outdent`) and keyboard shortcuts (e.g., `Tab`, `Shift-Tab`, `Backspace`). It ensures lists are handled separately using native commands. (`core-web/libs/block-editor/src/lib/extensions/indent/indent.extension.ts`)
* Updated the `DotBlockEditorComponent` to include the `IndentExtension` in its list of editor extensions. (`core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts`)
* Modified the `DotBubbleMenuPlugin` to incorporate the `IndentExtension` for non-list nodes while maintaining native commands for list items. (`core-web/libs/block-editor/src/lib/extensions/bubble-menu/plugins/dot-bubble-menu.plugin.ts`)

### Code Cleanup and Reorganization
* Consolidated and alphabetized import statements in multiple files for better readability. (`core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.ts`, `core-web/libs/block-editor/src/lib/extensions/bubble-menu/plugins/dot-bubble-menu.plugin.ts`) [[1]](diffhunk://#diff-266eab162f8661e695c3e40956692fa3696fbd4e8cd3e0352ea9e01f90b13609L36-R36) [[2]](diffhunk://#diff-c8d4bb53e36d994727c35eba8571ce71e300c27447a472e866cc6c01be20b101L31-R31)
* Updated the `index.ts` file to export the new `IndentExtension` and removed redundant exports for AI-related extensions. (`core-web/libs/block-editor/src/lib/extensions/index.ts`) [[1]](diffhunk://#diff-7c6595ce89739ae8d986dc0ee805e6cb6d00b0abcde4baec0479fa09e65e3823R2-R6) [[2]](diffhunk://#diff-7c6595ce89739ae8d986dc0ee805e6cb6d00b0abcde4baec0479fa09e65e3823L30-R35)

### Proposed Changes
- Updated import order in  for consistency.
- Added  to the block editor component.
- Enhanced  to handle indentation and outdentation for list items and other block types using the new .
- Cleaned up unused imports in

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots

https://github.com/user-attachments/assets/2b53a1ba-9882-4797-8535-018f0e3741ad



This PR fixes: #32106

oidacra committed:
refactor(block-editor): Reorganize imports and enhance bubble menu functionality

- Updated import order in  for consistency.
- Added  to the block editor component.
- Enhanced  to handle indentation and outdentation for list items and other block types using the new .
- Cleaned up unused imports in

nicobytes commented on core-web/libs/block-editor/src/lib/extensions/indent/indent.extension.ts:
```diff
@@ -0,0 +1,225 @@
+        return {
+            indent:
+                () =>
+                ({ tr, state, dispatch, editor }) => {
```
'tr' and editor are declared, but their value is never read; please just use state and dispatch in this case.

oidacra committed:
refactor(block-editor): Simplify indent and outdent functions in IndentExtension

- Refactored the `indent` and `outdent` methods to remove unnecessary parameters, improving code clarity.
- Ensured that the functionality remains focused on paragraphs, headings, and blockquotes, excluding lists.

oidacra commented on core-web/libs/block-editor/src/lib/extensions/indent/indent.extension.ts:
```diff
@@ -0,0 +1,225 @@
+}
+
+enum IndentProps {
+    min = 0,
+    max = 400, // Maximum limit of 400px
+    more = 40, // Increment of 40px
+    less = -40
```
Yes, that's a good idea, being more explicit.

oidacra committed:
refactor(block-editor): Replace enum with constants for indent properties

- Replaced the `IndentProps` enum with constant values for minimum, maximum, increment, and decrement indent values, enhancing code clarity and maintainability.
- Updated the `indent` and `outdent` functions to utilize the new constants, ensuring consistent behavior across the extension.
